docs/developer/release-process.md

```diff
@@ -30,8 +30,10 @@ To create a new release, follow these steps:
 3. Test the main branch for release-readiness. For that, use the `edge` containers, which are built from the main
    branch, and the [example applications](/examples).
 4. If a problem is found, prepare a fix PR, merge it into the main branch and return to the previous step.
-5. Create a release branch with a name that follows the `release-X.Y` format.
-6. Prepare and merge a PR into the release branch to update the repo files for the release:
+5. If the supported Gateway API minor version has changed since the last release, test NGF with the previous version of the Gateway API CRDs.
```

Comment:
```suggestion
5. If the supported Gateway API minor version has changed since the last release, test NGINX Gateway Fabric with the previous version of the Gateway API CRDs.
```